azext_iot/assets/user_messages.py

```diff
@@ -11,10 +11,13 @@
 
 
 def error_no_hub_or_login_on_input(entity_type="IoT Hub"):
-    return (
-        "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
-        " or {0} connection string via --login..."
-    ).format(entity_type, PARAM_MAPPING[entity_type])
+    if entity_type:
+        return (
+            "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
+            " or {0} connection string via --login..."
+        ).format(entity_type, PARAM_MAPPING[entity_type])
+    else:
+        return ("Required entity type missing, please check _validators.py for required entities.")
```

Comment:
Might be a nit - but I think I recall the CLI core linter using the pylint [`no_else_return`](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html) rule - basically we don't need the `else` here, since the code is either going to return inside of the `if`, or on the next line anyway.

This didn't seem to flag in our PR pipeline, so it might be a non-issue, but just a heads up in case we see it later.